tests/system_tests/test_plans_system.py

```diff
@@ -29,3 +33,13 @@ def test_spectroscopy(
         timeout=10,
     )
     assert events["FINISHED"][0]["scanDimensions"] == [5]
+
+
+def test_spectroscopy_re():
```

Comment:
Could: My only objection to putting tests like this back in is that I can't run them from my laptop over eduroam/vpn, which is very convenient. However if you think they add more value than take I won't block over this.

Reply:
I wrote this for two reasons:
- Blueapi was not working
- Passing the scanspec into the TaskRequest didn't seem to work. Do you have an example
  of that?

Reply:
I suspect the reason the spec didn't work is because there is no type bound in the plan signature:

```diff
def spectroscopy(
    spectroscopy_detector: AravisDetector = spectroscopy_detector,
    sample_stage: XYZStage = sample_stage,
-    spec: Spec | None = None,
+   spec: Spec[Movable] | None = None,
    exposure_time: float = 0.1,
    metadata: dict[str, Any] | None = None,
) -> MsgGenerator[None]:
```

Specifying a bluesky type tells blueapi to resolve it as a special case (device or devices). I also made a PR on top of this PR that just skips tests like this if you're not on a beamline machine, up to you. https://github.com/DiamondLightSource/test-rig-bluesky/pull/17/files

Reply:
I have pulled in the PR.

So if I add `Spec[Movable]` I can do something like this?

```python
    _sample_stage = sample_stage()
    scan_spec = Line(_sample_stage.x, 2, 5, 30) * Line(_sample_stage.y, 0, 5, 50)
    events = bluesky_plan_runner.run(
        TaskRequest(
            name="demo_spectroscopy",
            params={"spec": scan_spec.serialize()},
            instrument_session=latest_commissioning_instrument_session,
        ),
        timeout=10,
    )
```

I can't test currently because rabbit mq is not running.

Reply:
Yes, I believe that should work